### PR TITLE
[PCFG-119] Configuration keys

### PR DESCRIPTION
**Pull request by aomegax**
#### List of Changes
- Added functionalities for configuration keys entity:
    - get list of configuration entries
    - get details about a configuration entry
    - create a configuration entry
    - update a configuration entry
    - delete a configuration entry
- Updated Postman collection
- Updated Swagger 

#### How Has This Been Tested?

- unit test
- manual test

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: api-test/ApiConfig.postman_collection.json [hunk 1]
```diff
@@ -3779,6 +3779,438 @@
...
+									"  pm.expect(responseJson.config_description).to.be.exist;",
+									"});"
+								],
+								"type": "text/javascript"
+							}
+						}
+					],
+					"request": {
+						"method": "POST",
+						"header": [],
```
Comment (jacopocarlini) [inline api-test/ApiConfig.postman_collection.json:3844]: for k6 conversion it is better to make explicit the `Content-Type` header in POST and PUT operations.

File: src/main/java/it/pagopa/pagopa/apiconfig/controller/ConfigurationController.java [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+     * @return OK. (status code 200)
+     * or Service unavailable (status code 500)
+     */
+    @Operation(summary = "Get details of configuration key", security = {@SecurityRequirement(name = "ApiKey"), @SecurityRequirement(name = "Authorization")}, tags = {"Configuration"})
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "OK", content = @Content(mediaType = MediaType.APPLICATION_JSON_VALUE, schema = @Schema(implementation = ConfigurationKey.class))),
+            @ApiResponse(responseCode = "400", description = "Bad Request", content = @Content(mediaType = MediaType.APPLICATION_JSON_VALUE, schema = @Schema(implementation = ProblemJson.class))),
+            @ApiResponse(responseCode = "401", description = "Unauthorized", content = @Content(schema = @Schema())),
+            @ApiResponse(responseCode = "403", description = "Forbidden", content = @Content(schema = @Schema())),
+            @ApiResponse(responseCode = "429", description = "Too many requests", content = @Content(schema = @Schema())),
```
Comment (jacopocarlini) [inline src/main/java/it/pagopa/pagopa/apiconfig/controller/ConfigurationController.java:87]: missing 404

File: src/main/java/it/pagopa/pagopa/apiconfig/controller/ConfigurationController.java [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+     * @return OK. (status code 200)
+     * or Service unavailable (status code 500)
+     */
+    @Operation(summary = "Update configuration key", security = {@SecurityRequirement(name = "ApiKey"), @SecurityRequirement(name = "Authorization")}, tags = {"Configuration"})
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "201", description = "Created", content = @Content(mediaType = MediaType.APPLICATION_JSON_VALUE, schema = @Schema(implementation = ConfigurationKey.class))),
+            @ApiResponse(responseCode = "400", description = "Bad Request", content = @Content(mediaType = MediaType.APPLICATION_JSON_VALUE, schema = @Schema(implementation = ProblemJson.class))),
+            @ApiResponse(responseCode = "401", description = "Unauthorized", content = @Content(schema = @Schema())),
+            @ApiResponse(responseCode = "403", description = "Forbidden", content = @Content(schema = @Schema())),
+            @ApiResponse(responseCode = "409", description = "Conflict", content = @Content(mediaType = MediaType.APPLICATION_JSON_VALUE, schema = @Schema(implementation = ProblemJson.class))),
```
Comment (jacopocarlini) [inline src/main/java/it/pagopa/pagopa/apiconfig/controller/ConfigurationController.java:109]: PUT operations don't have 409, but 404

File: src/main/java/it/pagopa/pagopa/apiconfig/service/ConfigurationService.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * @param category configuration category
+     * @param key configuration key
+     * @return return the configuration key record from DB if exists
+     * @throws AppException if not found
+     */
+    protected it.pagopa.pagopa.apiconfig.entity.ConfigurationKeys getConfigurationKeyIfExists(String category, String key) throws AppException {
```
Comment (jacopocarlini) [inline src/main/java/it/pagopa/pagopa/apiconfig/service/ConfigurationService.java:86]: why is protected and not private?

File: src/main/java/it/pagopa/pagopa/apiconfig/service/ConfigurationService.java [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    }
+
+    public ConfigurationKey updateConfigurationKey(String category, String key, ConfigurationKey configurationKey) {
+        Optional<it.pagopa.pagopa.apiconfig.entity.ConfigurationKeys> configKey = configurationKeysRepository.findByConfigCategoryAndConfigKey(category, key);
+        if (!configKey.isPresent()) {
+            throw new AppException(AppError.CONFIGURATION_KEY_NOT_FOUND, category, key);
+        }
+
+        it.pagopa.pagopa.apiconfig.entity.ConfigurationKeys configKeyEntity = configKey.get();
+        configKeyEntity.setConfigValue(configurationKey.getConfigValue());
```
Comment (jacopocarlini) [inline src/main/java/it/pagopa/pagopa/apiconfig/service/ConfigurationService.java:56]: query the view is for the lack of id?

File: src/main/java/it/pagopa/pagopa/apiconfig/controller/ConfigurationController.java (hunk 3 above)
Comment (aomegax) [reply src/main/java/it/pagopa/pagopa/apiconfig/controller/ConfigurationController.java:109]: done

File: src/main/java/it/pagopa/pagopa/apiconfig/service/ConfigurationService.java (hunk 4 above)
Comment (aomegax) [reply src/main/java/it/pagopa/pagopa/apiconfig/service/ConfigurationService.java:86]: used a wrong shortcut, it is private

**Review by jacopocarlini — APPROVED** [review]
good job 👍

**Pull request merged**